### PR TITLE
Validation Tests for Apps

### PR DESCRIPTION
**Pull request by tiffanytfmao**

### CODE REVIEW

File: tests/validation/tests/v3_api/test_app.py [hunk 1]
```diff
@@ -1,5 +1,195 @@
...
+        targetNamespace=ns.name,
+        projectId=p.id,
+    )
+    app = proj_client.reload(app)
+    wait_for_app_to_active(proj_client, app.id)
+    admin_client.delete(p, ns, proj_client)
+    admin_client.delete(catalog1)
+    time.sleep(10)
+    assert len(admin_client.list_clusterCatalog()["data"]) == 0
+    try:
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: Add it to a function and call the function here instead.

File: tests/validation/tests/v3_api/test_app.py [hunk 2]
```diff
@@ -1,5 +1,195 @@
...
+def test_app_delete_project_catalog():
+    name = random_test_name()
+    admin_client = get_admin_client()
+    clusters = admin_client.list_cluster(name=CLUSTER_NAME).data
+    assert len(clusters) > 0
+    p, ns = create_project_and_ns(
+        ADMIN_TOKEN,
+        clusters[0],
+        "test-" + random_str())
+    url = "https://git.rancher.io/charts"
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: create a global variable

File: tests/validation/tests/v3_api/test_app.py [hunk 3]
```diff
@@ -1,5 +1,195 @@
...
+    client = get_admin_client()
+    catalog1 = client.create_clusterCatalog(name="clustercatalog",
+                                            baseType="clusterCatalog",
+                                            branch="master",
+                                            url=url,
+                                            clusterId=clusters[0].id
+                                            )
+    time.sleep(10)
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add a function to check if the catalog is Active

File: tests/validation/tests/v3_api/test_app.py [hunk 4]
```diff
@@ -1,5 +1,195 @@
...
+        externalId=proj_external_id,
+        answers=answers,
+        targetNamespace=ns.name,
+        projectId=p.id,
+    )
+    app = proj_client.reload(app)
+    wait_for_app_to_active(proj_client, app.id)
+    admin_client.delete(p, ns, proj_client)
+    admin_client.delete(catalog1)
+    time.sleep(10)
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add a function to check if the app is deleted

File: tests/validation/tests/v3_api/test_app.py [hunk 5]
```diff
@@ -1,5 +1,195 @@
...
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add a function to check if the app is deleted

File: tests/validation/tests/v3_api/test_app.py [hunk 6]
```diff
@@ -1,5 +1,195 @@
...
+    assert len(admin_client.list_projectCatalog()["data"]) == 0
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: Add it to a function and call the function here instead.

File: tests/validation/tests/v3_api/test_app.py [hunk 7]
```diff
@@ -122,3 +278,292 @@ def test_tiller():
...
+    projects = [[p, ns], [p2, ns2]]
+    url = "https://git.rancher.io/charts"
+    client = get_admin_client()
+    catalog1 = client.create_clusterCatalog(name="clustercatalog",
+                                            baseType="clusterCatalog",
+                                            branch="master",
+                                            url=url,
+                                            clusterId=clusters[0].id
+                                            )
+    time.sleep(10)
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add a function to check if the catalog is Active

File: tests/validation/tests/v3_api/test_app.py [hunk 8]
```diff
@@ -122,3 +278,292 @@ def test_tiller():
...
+    url = "https://git.rancher.io/charts"
+    client = get_admin_client()
+    catalog1 = client.create_projectCatalog(name="projectcatalog",
+                                            baseType="projectCatalog",
+                                            branch="master",
+                                            url=url,
+                                            projectId=p.id,
+                                            clusterId=clusters[0].id
+                                            )
+    time.sleep(10)
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add a function to check if the catalog is Active

File: tests/validation/tests/v3_api/test_app.py [hunk 9]
```diff
@@ -122,3 +278,292 @@ def test_tiller():
...
+    app = proj_client.create_app(
+        name=name,
+        externalId=wp_external_id,
+        targetNamespace=ns.name,
+        projectId=p.id,
+        answers=answers
+    )
+
+    app = proj_client.reload(app)
+    wait_for_app_to_active(proj_client, app.id)
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_app.py]: add "delete app"

**Comment by alena1108** [comment]
@sowmyav27 can you look into squashing the commits + addressing the remaining comments on top of that + publishing the changes as a part of a new pr.

**Comment by alena1108** [comment]
Closing as the changes will be incorporated in the new prs done by @sowmyav27

**Pull request closed without merging**